### DISCUSSION TITLE
Enrich against a custom .gmt

### DISCUSSION THREAD

**Discussion by apelin20** [egeulgen/pathfindR#197] [Ideas]
Hello, I was wondering if it's possible to enrich against a custom database, for example in .gmt format? There is an option to specify a custom PIN .sif file for subnetwork searches but I don't think I see custom databases. Would this be hard to implement? Thanks

**Comment by egeulgen** [comment]
there is an unexported function `pathfindR::gset_list_from_gmt(path2gmt="/path/to/gmt/file")` that returns a named list:

```r
custom_gene_sets <- pathfindR:::gset_list_from_gmt(path2gmt="/path/to/gmt/file")
```

then you can run pathfindR as follows:

```r
output_df <- run_pathfindR(input_df, gene_sets = "Custom", custom_genes = custom_gene_sets$genes_list, custom_descriptions = custom_gene_sets$descriptions_vec)
```
